content/blog/2018-09-14-smapr.md

```diff
@@ -0,0 +1,637 @@
+library(animation)
+library(patchwork)
+library(sf)
+
+worldmap <- getMap()
+study_area <- subset(worldmap, NAME_SORT %in% c('Somalia', 'Kenya'))
+study_area_sf <- as(study_area, 'sf')
+
+plot(worldmap)
+plot(study_area, add = TRUE, col = 'dodgerblue')
+```
+
+![](/img/blog-images/2018-09-14-smapr/map-aoi-1.png)<!-- -->
+
+## Finding soil moisture data
+
+Next, we can use the `find_smap` function to find global soil moisture data for any day.
+The SMAP satellite was launched in 2015 with two sensors: an active microwave sensor and a passive radiometer.
+The active sensor has since failed, so we will use the radiometer data, specifically version five of the 'SPL3SMP' data product (for a full list of data products see https://smap.jpl.nasa.gov/data/).
+
+
+```r
+find_smap('SPL3SMP', dates = '2018-03-01', version = 5)
+```
+
+```
+##                               name       date                     dir
+## 1 SMAP_L3_SM_P_20180301_R16010_001 2018-03-01 SPL3SMP.005/2018.03.01/
+```
+
+This returns a data frame with one row per file - we can see here that there is one file available for that date.
+If we wanted to search over a range of dates, we could provide a date sequence:
+
+
+```r
+date_seq <- seq(as.Date('2018-03-01'), as.Date('2018-03-06'), by = 1)
+files <- find_smap('SPL3SMP', dates = date_seq, version = 5)
+files
+```
+
+```
+##                               name       date                     dir
+## 1 SMAP_L3_SM_P_20180301_R16010_001 2018-03-01 SPL3SMP.005/2018.03.01/
+## 2 SMAP_L3_SM_P_20180302_R16010_001 2018-03-02 SPL3SMP.005/2018.03.02/
+## 3 SMAP_L3_SM_P_20180303_R16010_001 2018-03-03 SPL3SMP.005/2018.03.03/
+## 4 SMAP_L3_SM_P_20180304_R16010_001 2018-03-04 SPL3SMP.005/2018.03.04/
+## 5 SMAP_L3_SM_P_20180305_R16010_001 2018-03-05 SPL3SMP.005/2018.03.05/
+## 6 SMAP_L3_SM_P_20180306_R16010_001 2018-03-06 SPL3SMP.005/2018.03.06/
+```
+
+## Downloading and extracting soil moisture data
+
+The `download_smap` function takes the results from `find_smap` and downloads files locally:
+
+
+```r
+downloads <- download_smap(files, overwrite = FALSE)
+downloads
+```
+
+```
+##                               name       date                     dir
+## 1 SMAP_L3_SM_P_20180301_R16010_001 2018-03-01 SPL3SMP.005/2018.03.01/
+## 2 SMAP_L3_SM_P_20180302_R16010_001 2018-03-02 SPL3SMP.005/2018.03.02/
+## 3 SMAP_L3_SM_P_20180303_R16010_001 2018-03-03 SPL3SMP.005/2018.03.03/
+## 4 SMAP_L3_SM_P_20180304_R16010_001 2018-03-04 SPL3SMP.005/2018.03.04/
+## 5 SMAP_L3_SM_P_20180305_R16010_001 2018-03-05 SPL3SMP.005/2018.03.05/
+## 6 SMAP_L3_SM_P_20180306_R16010_001 2018-03-06 SPL3SMP.005/2018.03.06/
+##                   local_dir
+## 1 /home/rstudio/.cache/smap
+## 2 /home/rstudio/.cache/smap
+## 3 /home/rstudio/.cache/smap
+## 4 /home/rstudio/.cache/smap
+## 5 /home/rstudio/.cache/smap
+## 6 /home/rstudio/.cache/smap
+```
+
+Next, we can list the contents of the files we downloaded.
+These are HDF5 files, so each file contains multiple datasets.
```

Comment:
Can you link HDF5 for people unfamiliar to get more info? e.g. https://portal.hdfgroup.org/pages/viewpage.action?pageId=48804287 or https://en.wikipedia.org/wiki/Hierarchical_Data_Format

Reply:
Yep! https://github.com/ropensci/roweb2/pull/277/commits/969c944c405ea49113fcd1912fc368ea650d93c1